### ISSUE TITLE
GitLab IT `test_indexing` fails with stalled index in sandbox

### ISSUE THREAD

**Issue by achave11-ucsc** [DataBiosphere/azul#3835]
GitLab IT job fails `test_indexing` occasionally due to a stalled index job. This was [observed](https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/38892#L1902) in a recent sandbox deployment pipeline.
```
ERROR: test_indexing (integration_test.IndexingIntegrationTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/azul_test_case.py", line 176, in wrapped
    return method(*args, **kwargs)
  File "/builds/ucsc/azul/test/integration_test.py", line 353, in test_indexing
    _wait_for_indexer()
  File "/builds/ucsc/azul/test/integration_test.py", line 332, in _wait_for_indexer
    self.azul_client.wait_for_indexer()
  File "/builds/ucsc/azul/src/azul/azulclient.py", line 481, in wait_for_indexer
    self.queues.wait_to_stabilize()
  File "/builds/ucsc/azul/src/azul/queues.py", line 261, in wait_to_stabilize
    raise Exception('The queues have stalled', final_length)
Exception: ('The queues have stalled', 9)
```

**Comment by hannes-ucsc** [comment]
In this case the bundle was partitioned and nine of the 16 partitions failed with a TransportError about too many `bulk index` requests. Partitions are processed by the retry lambda so those nine failures sat in the retry queue for 15 min before being picked up again at which time they succeeded. We should be prepared for this. In the worst case, a stall could last 5 + 9 * 15 + 1 + 9 * 10 = 231 min (all lambda timeouts and their respective retries). That's obviously a bit long but I think the particular scenario we encountered here is very probable. The delay of the initial unpartitioned attempt is likely going to be occurring when other bundles are making progress and the queue lengths are fluctuating. The same goes for the first partitioned attempt (the one that failed for nine bundles). It will also be drowned out by other traffic in the queues. 

TL;DR We currently hardcode a stall timeout of 10 minutes. We should make that `max(config.contribution_lambda_timeout(retry=True), config.aggregation_lambda_timeout(retry=True))`.

**Comment by achave11** [comment]
This happened again https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/41190.

**Comment by achave11** [comment]
Another one https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/42052.

**Comment by achave11** [comment]
@hannes-ucsc: "This is currently not reproducible due to the substantial changes introduced by the fix for #2789. We're putting this on hold. As soon as it happens again we'll try to reproduce this locally and resume the work. Operator: please put this in triage as soon as this happens again on GitLab."

**Comment by achave11** [comment]
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/44802

**Comment by achave11** [comment]
@hannes-ucsc: "Assignee to attempt to reproduce the most resent failure posted by @achave11."

**Comment by achave11** [comment]
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/45248/raw

**Comment by hannes-ucsc** [comment]
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/46485

**Comment by danielsotirhos** [comment]
For demo, show a couple recent Gitlab job logs for sandbox or dev that successfully reached the "queues have emptied" state.

**Comment by melainalegaspi** [comment]
@hannes-ucsc : "Re-open if this occurs again."

**Comment by hannes-ucsc** [comment]
We either forgot to demo this or, if we did, forgot to label it correctly.
